lib/msal-common/src/authority/Authority.ts

```diff
@@ -543,17 +544,25 @@ export class Authority {
         let match = null;
         try {
             const response =
-                await this.networkInterface.sendGetRequestAsync<CloudInstanceDiscoveryResponse>(
+                await this.networkInterface.sendGetRequestAsync<CloudInstanceDiscoveryResponse | CloudInstanceDiscoveryErrorResponse>(
                     instanceDiscoveryEndpoint,
                     options
                 );
-            const metadata = isCloudInstanceDiscoveryResponse(response.body)
-                ? response.body.metadata
-                : [];
-            if (metadata.length === 0) {
-                // If no metadata is returned, authority is untrusted
-                return null;
+            
+            let typedResponseBody: CloudInstanceDiscoveryResponse | CloudInstanceDiscoveryErrorResponse;
```

Comment:
This block needs more logs.

Reply:
I've added more logs in the other PR that is specifically for logging.

Reply:
+1. The logs should cover all the scenarios in this class - why certain code path was taken (if it's not already logged in other classes like Authority). For ex: log why we return null.